test/runtests.jl

```diff
@@ -13,10 +13,8 @@ include("test_utils.jl")
 @testset verbose=true "NNlib.jl" begin
     if CUDA.functional()
         if get(ENV, "NNLIB_TEST_CUDA", "false") == "true"
-            import Pkg
-            using NNlibCUDA
             @testset "CUDA" begin
-                Pkg.test("NNlibCUDA")
+                include("cuda/runtests.jl")
```

Comment:
Change the folder structure to 
```julia
include("ext/cuda/runtests.jl")
```
?

Reply:
The issue is that Pkg expects a flat structure for each extension (i.e. `ext/CUDAExt` is the `src/` folder), so there's no good place for tests. Ref. https://pkgdocs.julialang.org/dev/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions)

Reply:
Wait is the suggestion `/test/ext/cuda/runtests.jl` or something like `/ext/CUDAExt/...`? 

The former just seems more verbose.

The latter is a bit messy as there's no `/ext/CUDAExt/src/` folder. Unlike independent packages in the same repository. Maybe that's intentional --- extensions don't have an independent existence, so don't really deserve tests you can run separately?

Reply:
`ext/CUDAExt` is required for the extension mechanism to work, AIUI. So if we were to move the tests into the extension dir, they'd have to be under `ext/CUDAExt/tests/...` and that's (as you note) messy.

Reply:
The suggestion is to do create an `ext` folder inside the `test` ones, so have `/test/ext/cuda/runtests.jl`instead of `/test/cuda/runtests.jl`, this also considering we will likely have other extensions in the future. Not really important though.

Reply:
Ok. It doesn't seem awful to have say `/test/cuda/` and `/test/amd/` at top level, without organising these into another layer of hierarchy. 

They could be named `/test/CUDAext/` etc to stress the connection I guess.

Reply:
Seems like this is something we can put a pin in for the day when we get extensions which aren't just backends.